src/main/java/net/fabricmc/loom/configuration/providers/mappings/TinyMappingsService.java

```diff
@@ -31,22 +31,24 @@
 import net.fabricmc.loom.util.service.SharedService;
 import net.fabricmc.loom.util.service.SharedServiceManager;
 import net.fabricmc.mappingio.MappingReader;
+import net.fabricmc.mappingio.adapter.MappingSourceNsSwitch;
 import net.fabricmc.mappingio.tree.MemoryMappingTree;
 
 public final class TinyMappingsService implements SharedService {
 	private final MemoryMappingTree mappingTree;
 
-	public TinyMappingsService(Path tinyMappings) {
+	public TinyMappingsService(Path tinyMappings, String srcNs) {
```

Comment:
I have removed (Ill push my changes shortly) the srcNs from here as I dont think its needed or ocrrect. The source NS is controlled by the input mapping file, this service is basically a shared way to read the raw mapping file, if another NS is needed it should be handled by the consumer of this service.

Reply:
I added that because of [this error](https://github.com/SpaceWalkerRS/fabric-loom/blob/05fbc2d539012e4d9eea59270e3ba291fbe14d51/src/main/java/net/fabricmc/loom/util/TinyRemapperHelper.java#L72), which otherwise causes a crash. Should there be a SourceNsSwitch there instead of an exception then?

Reply:
nvm I see you change the fixRecords flag based on the MC provider - but the LegacyMergedMinecraftProvider uses delegate single jar providers, so it doesn't work

Reply:
Ill take a look, can maybe improve this to not depend on the provider at all.

Reply:
looking more into this, it's basically every place where you replaced `minecraftProvider.canMergeJars()` with `!(minecraftProvider instanceof LegacyMergedMinecraftProvider)` that is broken now, and that's quite a few places.

Reply:
that might be partially 'cause I was using a server only test project, but those should still use the new namespaces and whatnot in pre-1.3 versions

Reply:
Yeah, I have just noticed the same, I might revert this 🤔